src/Signature/SignatureV4.php

```diff
@@ -43,9 +50,13 @@ public function signRequest(
         if ($token = $credentials->getSecurityToken()) {
             $parsed['headers']['X-Amz-Security-Token'] = [$token];
         }
-
         $cs = $this->createScope($sdt, $this->region, $this->service);
         $payload = $this->getPayload($request);
+
+        if ($this->unsigned) {
```

Comment:
using `$this->unsigned` appears simple at first glance, yet as you have also mentioned, this check in targeting at whether a request is valid for unsigned payload, so a complete check is checking the boolean and scheme as same time, which is already done so in the `getPayload`. So even this works, this check is incomplete, and a complete check will duplicate the check [here](https://github.com/aws/aws-sdk-php/pull/1206/files#diff-50459fdb298219f1d103da728ab4afb2R142). That's why I'm suggesting checking payload value directly.